relevanceai/http_client.py

```diff
@@ -100,3 +101,32 @@ def check_auth(self):
             return response.status_code == 200    
         except:
             raise Exception("Invalid auth details.")
+
+    def kmeans_cluster(self, dataset_id, vector_fields, k = 10, update_dataset = True):
+        # load the documents
+        # ToDo: tqdm for loading data
+        docs = []
+        batch = self.datasets.documents.list(dataset_id, select_fields = vector_fields)
+        cursor = batch['cursor']
+        while batch:
+            docs.extend(batch['documents'])
+            batch = self.datasets.documents.list(dataset_id, select_fields = vector_fields, cursor = cursor)
+
+        # Cluster
+        clusterer = KMeans(k=k)
+        clustered_docs = clusterer.fit_documents(vector_fields, docs, return_only_clusters=True)
+
+        # Write back the results
+        if update_dataset:
```

Comment:
I think it should always `update_dataset` shouldn't it?

Reply:
I just thought maybe we want to add an option to keep the original db as it is and create a copy with the clustering information. But it slows down the process for big databases, at the same time it is not a priority at all.